packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,182 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the appNotification specific part of the SDK
+ */
+export namespace appNotification {
+  /**
+   * Data structure to represent appNotification information
+   */
+  export interface NotificationDisplayParam {
+    /**
+     * Notification title(maximum length: 20 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 40 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: string;
```

Comment:
>string [](http://example.com/codeflow?start=11&length=6)

Since you are able to use the `URL` type in this interface (i.e. you use it for the notification action URL), should this also be a `URL` then?

Reply:
Yes, that sounds like a better idea!
I've implemented that change.